File: TripKitAndroid/src/main/java/com/skedgo/tripkit/tsp/RegionInfoBody.java
package com.skedgo.tripkit.tsp;

import com.google.gson.annotations.JsonAdapter;
import com.skedgo.tripkit.common.model.Region;

import org.immutables.value.Value;

import static org.immutables.gson.Gson.TypeAdapters;
import static org.immutables.value.Value.Immutable;
import static org.immutables.value.Value.Style;

@Immutable
@TypeAdapters
@Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersRegionInfoBody.class)
public interface RegionInfoBody {
    /**
     * @return {@link Region#getName()}.
     */
    @Value.Parameter
    String region();
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/tsp/RegionInfoBody.kt
package com.skedgo.tripkit.tsp

import com.google.gson.annotations.JsonAdapter
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Parameter
import org.immutables.value.Value.Style

@Immutable
@TypeAdapters
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersRegionInfoBody::class
)
interface RegionInfoBody {
    /**
     * @return [Region.getName].
     */
    @Parameter
    fun region(): String
}